code/modules/projectiles/gun_helpers.dm

```diff
@@ -563,7 +563,14 @@ As sniper rifles have both and weapon mods can change them as well. ..() deals w
 		return
 	balloon_alert(usr, "No usable underrail attachments")
 
+///Toggles weapons ejecting their magazines when they're empty. This one is one a gun level and is used via right clicking the gun.
+/obj/item/weapon/gun/verb/toggle_auto_eject()
+	set category = null
+	set name = "Toggle Automatic Magazine Ejection (Weapon)"
+	set desc = "Toggles the gun automatically unloading it's magazine upon depletion of magazine."
```

Comment:
'it's' is the contraction of 'it is', 'its' is the possessive. Also 'unloading its magazine upon depletion of magazine' sounds weird.

Reply:
This is probably never going to be seen but yeah.